databroker/v2.py

```diff
@@ -13,13 +13,58 @@ class Broker(Catalog):
     This is a thin wrapper around intake.Catalog.
 
     It includes an accessor the databroker API version 1.
+
+    Parameters
+    ----------
+    handler_registry : dict, optional
+        This is passed to the Filler or whatever class is given in the
+        filler_class parametr below.
+
+        Maps each 'spec' (a string identifying a given type or external
+        resource) to a handler class.
+
+        A 'handler class' may be any callable with the signature::
+
+            handler_class(resource_path, root, **resource_kwargs)
+
+        It is expected to return an object, a 'handler instance', which is also
+        callable and has the following signature::
+
+            handler_instance(**datum_kwargs)
+
+        As the names 'handler class' and 'handler instance' suggest, this is
+        typically implemented using a class that implements ``__init__`` and
+        ``__call__``, with the respective signatures. But in general it may be
+        any callable-that-returns-a-callable.
+    root_map: dict, optional
+        This is passed to Filler or whatever class is given in the filler_class
+        parameter below.
+
+        str -> str mapping to account for temporarily moved/copied/remounted
+        files.  Any resources which have a ``root`` in ``root_map`` will be
+        loaded using the mapped ``root``.
+    filler_class: type
+        This is Filler by default. It can be a Filler subclass,
+        ``functools.partial(Filler, ...)``, or any class that provides the same
+        methods as ``DocumentRouter``.
+    transforms : dict
+        A dict that maps any subset of the keys {start, stop, resource, descriptor}
+        to a function that accepts a document of the corresponding type and
+        returns it, potentially modified. This feature is for patching up
+        erroneous metadata. It is intended for quick, temporary fixes that
+        may later be applied permanently to the data at rest
+        (e.g via a database migration).
+    **kwargs :
+        Additional keyword arguments are passed through to the base class,
+        Catalog.
     """
 
     def __init__(self, *, handler_registry=None, root_map=None,
-                 filler_class=event_model.Filler, **kwargs):
+                 filler_class=event_model.Filler, transforms=None, **kwargs):
 
         self._root_map = root_map
         self._filler_class = filler_class
+        self._transforms = parse_transforms(transforms)
```

Comment:
Probably it should be mentioned in the docstring that the transforms are in the string form.

Reply:
To be clear they _may_ be in the string form. Both a callable and a (string) import path to a callable are accepted. I agree we should mention that in the docstring.